js/client-battle.js

```diff
@@ -396,14 +396,15 @@
 						if (move.id === 'Struggle' || move.id === 'Recharge') pp = '&ndash;';
 						if (move.id === 'Recharge') move.type = '&ndash;';
 						if (name.substr(0, 12) === 'Hidden Power') name = 'Hidden Power';
+						var moveType = this.getMoveType(move, switchables[pos]);
```

Comment:
I still think it'd be better if for moves that vary, you got everything you knew about the move. Like, for moves with variable base power, you could display what base power it could be.

You could do that as two functions, though, in which case you wouldn't need to clone the object. Thoughts?